### PR TITLE
[Connectors API] Add set connector sync job error API docs

### PR DESCRIPTION
**Pull request by timgrein**
Add docs for the `PUT _connector/_sync_job/connector-sync-job-id/_error` API endpoint.

### CODE REVIEW

**Comment by timgrein** [comment]
@elasticmachine update branch

File: docs/reference/connector/apis/set-connector-sync-job-error-api.asciidoc [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+==== {api-prereq-title}
+
+* To sync data using connectors, it's essential to have the Elastic connectors service running.
+* The `connector_sync_job_id` parameter should reference an existing connector sync job.
+
+[[set-connector-sync-job-error-api-desc]]
+==== {api-description-title}
+
+Sets the `error` field for the specified connector sync job and sets its `status` to `error`.
+
```
Comment (abdonpijpelink) [inline docs/reference/connector/apis/set-connector-sync-job-error-api.asciidoc:23]:
```suggestion

[[get-connector-sync-job-api-path-params]]
==== {api-path-parms-title}

`<connector_sync_job_id>`::
(Required, string)

```

File: docs/reference/connector/apis/set-connector-sync-job-error-api.asciidoc [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+[[set-connector-sync-job-error-api-prereqs]]
+==== {api-prereq-title}
+
+* To sync data using connectors, it's essential to have the Elastic connectors service running.
+* The `connector_sync_job_id` parameter should reference an existing connector sync job.
+
+[[set-connector-sync-job-error-api-desc]]
+==== {api-description-title}
+
+Sets the `error` field for the specified connector sync job and sets its `status` to `error`.
```
Comment (abdonpijpelink) [inline docs/reference/connector/apis/set-connector-sync-job-error-api.asciidoc:22]: Maybe it's worthwhile explaining why someone would want to do this? (It's not clear to me)

**Review by abdonpijpelink — COMMENTED** [review]
Left a comment and a suggestion.

File: docs/reference/connector/apis/set-connector-sync-job-error-api.asciidoc (hunk 2 above)
Comment (timgrein) [reply docs/reference/connector/apis/set-connector-sync-job-error-api.asciidoc:22]: This is mainly an API for the connectors service (or any other system executing sync jobs), which need to set errors. I can add a paragraph that this is not expected to be used by end-users maybe?

Comment (abdonpijpelink) [reply docs/reference/connector/apis/set-connector-sync-job-error-api.asciidoc:22]: Yeah, that could be helpful.

**Review by abdonpijpelink — APPROVED** [review]

File: docs/reference/connector/apis/set-connector-sync-job-error-api.asciidoc [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+[[get-connector-sync-job-api-path-params]]
```
Comment (abdonpijpelink) [inline docs/reference/connector/apis/set-connector-sync-job-error-api.asciidoc:24]: Sorry, I made a mistake in my earlier suggestion which leads to a duplicate ID. This should fix it.
```suggestion
[[get-connector-sync-job-error-api-path-params]]
```

Comment (timgrein) [reply docs/reference/connector/apis/set-connector-sync-job-error-api.asciidoc]: No worries! Thank you for fixing :)

**Pull request merged**